scheds/include/scx/enums.bpf.h

```diff
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Convenience macros for getting/setting struct scx_enums instances.
+ *
+ * Copyright (c) 2024 Meta Platforms, Inc. and affiliates.
+ */
+#ifndef __SCX_ENUMS_BPF_H
+#define __SCX_ENUMS_BPF_H
+
+#include <scx/enums.h>
+
+#define SCX_ENUM_DEFINE() \
+	struct scx_enums _scx_enum;
+
+#define SCX_ENUM(name) \
+	(_scx_enum.SCX_##name)
```

Comment:
This makes each enum access a memory dereference. If you make each enum a `const volatile` variable and set them from the userspace side during init, then the verifier will handle them as consts and we can init them as a part of the `SCX_OPS_LOAD()` or `SCX_OPS_ATTACH()`.

For example, there can be `const volatile u64 __SCX_ENQ_WAKEUP;` which is initialized during load and then `#define SCX_ENQ_WAKEUP __SCX_ENQ_WAKEUP` would make it transparent for the scheduler implementations.

Reply:
I tried setting it that way, but there was the issue where the userspace component didn't have access to the ```vmlinux.h``` definitions and so could not set the initial value of the variables. I tried including ```vmlinux.h``` and got a bunch of errors but they must have been due to improperly structuring the header includes - I will go back to that and update.

Reply:
I think it's fine to fail loading if the enums are not available through BTF.